pkg/basecluster/validate_test.go

```diff
@@ -0,0 +1,103 @@
+package basecluster
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+)
+
+type testEntry struct {
+	DirName     string
+	ErrPattern  string
```

Comment:
same here too, we can make this type `error` and set the expected error to be nil for no errors otherwise just set them to the predefined errors. This could simplify checking for errors

Reply:
I have made the refactors here https://github.com/arlonproj/arlon/pull/new/private/Rohitrajak1807/unit-tests
 (was getting very hard to do it in the github review UI) please feel free to take a look.

Reply:
See my previous comments.